server/broadcast.go

```diff
@@ -434,44 +427,42 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 			// preemptively fetch segments, they could be reading tampered
 			// data. Not an issue if the delivery protocol is being obeyed.
 			segHashLock.Lock()
-			segURLs[i] = url
+			tr.url[i] = url
 			segHashLock.Unlock()
 
 			if monitor.Enabled {
 				monitor.TranscodedSegmentAppeared(nonce, seg.SeqNo, sess.Profiles[i].Name)
 			}
 		}
 
+		wg.Add(len(res.Segments))
 		for i, v := range res.Segments {
+			tr.url[i] = v.Url
 			go dlFunc(v.Url, v.Pixels, i)
 		}
 
-		cond.L.Lock()
-		for n != 0 {
-			cond.Wait()
-		}
-		cond.L.Unlock()
+		wg.Wait()
 		if dlErr != nil {
-			return dlErr
+			return nil, dlErr
 		}
 
 		if verifier != nil {
-			err := verify(verifier, cxn, sess, seg, res.TranscodeData, segURLs)
+			err := verify(verifier, cxn, sess, seg, res.TranscodeData, tr.url)
 			if err != nil {
 				glog.Errorf("Error verifying nonce=%d manifestID=%s seqNo=%d err=%s", nonce, cxn.mid, seg.SeqNo, err)
-				return err
+				return tr, err
```

Comment:
Is there a reason we return `tr` here but not for other errors?

Reply:
fixed